web/packages/extension/src/popup.ts

```diff
@@ -60,8 +61,17 @@ async function queryTabStatus(
             type: "ping",
         });
     } catch (e) {
-        listener("status_result_protected");
-        reloadButton.disabled = true;
+        const url = activeTab.url ? new URL(activeTab.url) : null;
```

Comment:
Nice improvement, thank you!
Can you please place the added logic outside the `catch`? My intuition here is that it doesn't have to do with any error-handling at all. I think it better fits right after `activeTab` is assigned.

Reply:
This definitely makes sense. I did the changes.